### ISSUE TITLE
Fill Factor

### ISSUE THREAD

**Issue by pveres** [testMOPS/mops#9]
@mjuric @ivezic @stevechesley  @rhiannonlynne 

In Seattle we had a discussions about the fill factor and how it is computed. And how does the actual area of pixels look like. 

Here, I attach a simple schematics of the focal plane. The dark blue circle is the area that we currently use for our FOV. I believe that the read area (thick red line) represents the actual pixels. The question is - if we use the actual area with the pixels - what is the fill factor (I mean area not used within the red-bounded chips)? This fill factor will consist of gaps and lines between chips and cells (if the cell is defined as 9x9 chip array). Plus, there might be additional fill factor from bright stars, etc.

![lsst_focalplane](https://cloud.githubusercontent.com/assets/13923689/10112682/006f3bb4-6392-11e5-96e9-d75d894dfad6.jpg)

For the circle, we used fill factor of 0.88. But the confusion was whether it was inner (most likely - this was chip gaps plus sensor-chips in the corrners.) or outer circle.

I know that somewhere in the MafSSO there is model for this configuration of camera. We found it at the bottom of Early Prototype SSO detections.ipynb. Is it possible to construct an array of good and bad chips or pixels (x,y)? We have it in MOPS including the camera rotation

**Comment by rhiannonlynne** [comment]
I have a notebook that can help investigate this. I will add it to our github repo here and link to this issue.
The short answer is that the inner circle has a fill factor of 89% (not including any effects from bright stars). The fill factor for the red outline is similar, but I haven't calculated it. 

I think I can construct an array of good/bad pixels or chips, but I'm not entirely clear what you need. Are you thinking something like "this x/y in the focal plane (in mm?) is the center of a chip (or raft) and the active space around it is dx/dy"? Or something like the location of all the corners of all the chips?

**Comment by pveres** [comment]
I have 2 ideas:
a) construct (5x5) = 25 cells that correspond to a square with the circle inside and cut-off 4 corners. Actually, the square will be the FOV and the mask will consist of 4 corners (4 cells). This will be the active area. Then, I will apply the fill factor for the visible area (by random numbers and fading function).

b) I can use your pattern. I need a mask that will determine, for every pixel, that this is visible (1) or behind a mask (0). PS MOPS has a mask that consist of 100x100 cells. Each cell is either visible (1) or masked (0). Plus you can rotate it by position angle w/r to RA, DEC. I can put the mask on top of the FOV (which can be either a square or a circle).

**Comment by rhiannonlynne** [comment]
Notebook calculating fill factor is added: 
https://github.com/testMOPS/mops/blob/master/FillFactor.ipynb

Generating a 100x100 mask version of that certainly can be done, might be the best option. Not sure how well it will capture the overall fill factor, given that we have pretty small gaps within the silicon, but we can see.

**Comment by rhiannonlynne** [comment]
We discussed this in person in Seattle, and the decision was to create a 9x9 mask, where each element is equivalent to a raft. The corners would be dropped, as these are engineering sensors only. 
The fill factor within each raft can be calculated, which we did in Seattle, as well as the sizes and location of the rafts. The calculation is captured here: https://github.com/rhiannonlynne/318-proceedings/blob/master/FillFactor.ipynb

ccd x/y 800.0 814.4
ccd gap x/y 45 30.6
raft x/y (w/o raft gap) 2490.0 2504.4
raft gap x/y 50 35.6
ccd and raft areas 651520.0 6235956.0
fill factor, without raft gaps 94.0301695522
total x/y (including outside gaps) 12700.0 12700.0
square side length 12700.0
raft xy with raft gaps 2540.0
fill factor, with raft gaps 90.8872217744
